sunkit_image/tests/test_offlimb.py

```diff
@@ -0,0 +1,138 @@
+    test_map2 = sunpy.map.Map((test_data2, header))
+    return test_map2
+
+
+@pytest.fixture
+def radial_bin_edges():
+    radial_bins = utils.equally_spaced_bins(inner_value=0.001, outer_value=0.003, nbins=5)
+    radial_bins = radial_bins * u.R_sun
+    return radial_bins
+
+
+def set_attenuation_coefficients(order):
+    attenuation_coefficients = np.zeros((2, order + 1))
+    attenuation_coefficients[0][:] = np.linspace(1, 0, order + 1)
+    attenuation_coefficients[1][:] = np.linspace(1, 0, order + 1)
+    return attenuation_coefficients
+
+
+def test_nrgf(map_test1, map_test2, radial_bin_edges):
+
+    result = np.zeros_like(map_test1.data)
+    expect = off.nrgf(map_test1, radial_bin_edges, application_radius=0.001*u.R_sun)
+
+    assert np.allclose(expect.data.shape, map_test1.data.shape)
+    assert np.allclose(expect.data, result)
+
+    result = [[0.,  1.,  0., -1., 0.],
+              [1.,  0.,  0.,  0., -1.],
+              [0.,  0.,  0.,  0.,  0.],
+              [1.,  0.,  0.,  0., -1.],
+              [0.,  1.,  0., -1., 0.]]
+
+    expect = off.nrgf(map_test2, radial_bin_edges, application_radius=0.001*u.R_sun)
+
+    assert np.allclose(expect.data.shape, map_test2.data.shape)
+    assert np.allclose(expect.data, result)
+
+
+def test_fnrgf(map_test1, map_test2, radial_bin_edges):
+
+    order = 1     
+    # TODO : Write few more tests with different order
+
+    result = [[-0.,  96., 128.,  96.,  -0.],
+              [96., 224., 288., 224.,  96.],
+              [128., 288.,   0., 288., 128.],
+              [96., 224., 288., 224.,  96.],
+              [-0.,  96., 128.,  96.,  -0.]]
+    attenuation_coefficients = set_attenuation_coefficients(order)
+    expect = off.fnrgf(map_test1, radial_bin_edges, order,
+                                                            attenuation_coefficients,
+                                                            application_radius=0.001*u.R_sun,
+                                                            number_angular_segments=4)
+    assert np.allclose(expect.data.shape, map_test1.data.shape)
+    assert np.allclose(expect.data, result)
+
+    result = [[-0., 128., 128.,  96.,  -0.],
+              [128., 224., 288., 224.,  96.],
+              [128., 288.,   0., 288., 128.],
+              [128., 224., 288., 224.,  96.],
+              [-0., 128., 128.,  96.,  -0.]]
+    expect = off.fnrgf(map_test2, radial_bin_edges, order,
+                                                            attenuation_coefficients,
+                                                            application_radius=0.001*u.R_sun,
+                                                            number_angular_segments=4)
+    assert np.allclose(expect.data.shape, map_test2.data.shape)
+    assert np.allclose(expect.data, result)
+
+
+@pytest.fixture
+@pytest.mark.remote_data
+def smap():    
+    return sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
```

Comment:
We should move to using the sunpy test data for the figures. To avoid the sample data and the online requirement.